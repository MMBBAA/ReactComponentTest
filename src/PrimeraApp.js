import React/*,{ Children }*/ from 'react';
import PropTypes from 'prop-types';

//import React, { Fragment } from 'react';
//FC
const PrimeraApp = ({ saludo, subtitulo }) => {



    return (
        <>
            <h1>{saludo}</h1>
            {/*  <pre>{JSON.stringify(saludo,null,3)}</pre>*/}
            <p>{subtitulo}</p>
        </>
    );
}

PrimeraApp.propTypes = {
    saludo: PropTypes.string.isRequired,
   // otraPropiedad: PropTypes.number.isRequired
}

PrimeraApp.defaultProps = {
    saludo:'Recurso:',
    subtitulo: 'calculadora de suma',
}

export default PrimeraApp;