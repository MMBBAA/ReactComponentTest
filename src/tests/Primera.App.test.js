import React from 'react';
import '@testing-library/jest-dom';
//import { render } from '@testing-library/react';
import PrimeraApp from '../PrimeraApp';
import { shallow } from 'enzyme';


describe(' Pruebas en <PrimeraApp/> ', () => {
    /*test('debe de mostrar el mensaje "Hola, Soy Juan "', () => {
        const saludo = "Hola, Soy Juan";
        const { getByText } = render(<PrimeraApp saludo={saludo} />);
        expect ( getByText( saludo ) ).toBeInTheDocument();
    })*/
    //prueba con Enzyme
    test('debe de mostrar <PrimeraApp /> correctamente', () => {

        //shallow lo importamos de Enzyme
        const saludo='Hola, Soy Juan';
        const wrapper=shallow(<PrimeraApp saludo={saludo}/>)

        expect(wrapper).toMatchSnapshot();
    });
    test('debe de mostrar el subtÃ­tulo enviado por props', () => {

        const saludo='Hola, Soy Juan';
        const subTitulo='Soy un subtitulo';
        const wrapper=shallow(
        <PrimeraApp saludo={saludo} 
        subtitulo={subTitulo}
        />
        );
            const textoParrafo=wrapper.find('p').text();
        //    console.log(textoParrafo);
    
        expect(textoParrafo).toBe(subTitulo);
    })
    
});