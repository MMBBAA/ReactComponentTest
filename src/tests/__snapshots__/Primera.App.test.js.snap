// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` Pruebas en <PrimeraApp/>  debe de mostrar <PrimeraApp /> correctamente 1`] = `
<Fragment>
  <h1>
    Hola, Soy Juan
  </h1>
  <p>
    calculadora de suma
  </p>
</Fragment>
`;

exports[` Pruebas en <PrimeraApp/>  debe de mostrar el subtítulo  1`] = `undefined`;

exports[` Pruebas en <PrimeraApp/>  debe de mostrar el subtítulo enviado por props 1`] = `undefined`;
